src/laddie/laddie_utilities.f90

```diff
@@ -69,68 +166,39 @@ SUBROUTINE allocate_laddie_model( mesh, laddie)
     CALL init_routine( routine_name)
 
     ! Thickness
-    ALLOCATE( laddie%H                  ( mesh%vi1:mesh%vi2              )) ! [m]             Layer thickness
-    ALLOCATE( laddie%H_prev             ( mesh%vi1:mesh%vi2              )) ! [m]             previous timestep
-    ALLOCATE( laddie%H_next             ( mesh%vi1:mesh%vi2              )) ! [m]             next timestep   
     ALLOCATE( laddie%dH_dt              ( mesh%vi1:mesh%vi2              )) ! [m]             change
 
-    laddie%H              = 0._dp
-    laddie%H_prev         = 0._dp
-    laddie%H_next         = 0._dp
     laddie%dH_dt          = 0._dp
 
-    ! Velocities
-    ALLOCATE( laddie%U                  ( mesh%ti1:mesh%ti2              )) ! [m s^-1]        2D velocity
-    ALLOCATE( laddie%V                  ( mesh%ti1:mesh%ti2              )) ! [m s^-1]  
-    ALLOCATE( laddie%U_prev             ( mesh%ti1:mesh%ti2              )) ! [m s^-1]        2D velocity previous timestep
-    ALLOCATE( laddie%V_prev             ( mesh%ti1:mesh%ti2              )) ! [m s^-1]  
-    ALLOCATE( laddie%U_next             ( mesh%ti1:mesh%ti2              )) ! [m s^-1]        2D velocity next timestep
-    ALLOCATE( laddie%V_next             ( mesh%ti1:mesh%ti2              )) ! [m s^-1]  
-
-    laddie%U              = 0._dp
-    laddie%V              = 0._dp
-    laddie%U_prev         = 0._dp
-    laddie%V_prev         = 0._dp
-    laddie%U_next         = 0._dp
-    laddie%V_next         = 0._dp
-
     ! Temperatures
-    ALLOCATE( laddie%T                  ( mesh%vi1:mesh%vi2              )) ! [degC]          Temperature
-    ALLOCATE( laddie%T_prev             ( mesh%vi1:mesh%vi2              )) ! [degC]          Temperature previous timestep
-    ALLOCATE( laddie%T_next             ( mesh%vi1:mesh%vi2              )) ! [degC]          Temperature next timestep
     ALLOCATE( laddie%T_amb              ( mesh%vi1:mesh%vi2              )) ! [degC]          Temperature layer bottom
     ALLOCATE( laddie%T_base             ( mesh%vi1:mesh%vi2              )) ! [degC]          Temperature ice shelf base
     ALLOCATE( laddie%T_freeze           ( mesh%vi1:mesh%vi2              )) ! [degC]          Temperature freezing
 
-    laddie%T              = 0._dp
-    laddie%T_prev         = 0._dp
-    laddie%T_next         = 0._dp
     laddie%T_amb          = 0._dp
     laddie%T_base         = 0._dp
     laddie%T_freeze       = 0._dp
```

Comment:
Shorter would be to say ALLOCATE( laddie%T( mesh%vi1:mesh%vi2), source = 0._dp).